src/SFA.DAS.Provider.PR.Web.UnitTests/Controllers/ContactDetails/ContactDetailsControllerPostTests.cs

```diff
@@ -88,6 +89,174 @@ public void Post_NamesAreTrimmed(
             (x => x.LastName == lastName)), Times.Once);
     }
 
+    [Test, MoqAutoData]
+    public void Post_FirstNameIsNull_ReturnsValidationError(
+        [Frozen] Mock<IValidator<ContactDetailsSubmitModel>> validatorMock,
+        [Frozen] Mock<ISessionService> sessionServiceMock,
+        [Greedy] ContactDetailsController sut,
+        int ukprn,
+        ContactDetailsSubmitModel contactDetailsSubmitModel,
+        CancellationToken cancellationToken
+    )
+    {
+        contactDetailsSubmitModel.FirstName = null;
+
+        sessionServiceMock.Setup(s => 
+            s.Get<AddEmployerSessionModel>()
+        ).Returns(
+            new AddEmployerSessionModel { Email = Email }
+        );
+
+        var validationFailures = new List<ValidationFailure>
+        {
+            new("FirstName", "FirstName field is invalid") { ErrorCode = "1001" }
```

Comment:
convert the magic string ('Firstname field is invalid") to a const as top and make it generic (eg ValidationErrorMessage="validation error") and use that in setup and assertion

Reply:
If it does not matter what the error is, then create this using AutoData attribute